services/ingestion/etl/bulgaria/xls/src/lib/transform.js

```diff
@@ -0,0 +1,252 @@
+import sanitizeBudgetItem from '@eubfr/lib/budget/budgetFormatter';
+import numeral from 'numeral';
+import type { Project } from '@eubfr/types';
+
+/**
+ * Preprocess `budget` field.
+ *
+ * Input fields taken from the `record` are:
+ * - `Total Eligible Costs Granted for the Operation`
+ * - `% of EC co-financing`
+ *
+ * @memberof BulgariaXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Budget}
+ */
+const getBudget = record => {
+  const { _value: total } = numeral(
+    record['Total Eligible Costs Granted for the Operation']
+  );
+  const { _value: percentage } = numeral(record['% of EC co-financing']);
+  const euContrib = (total * percentage) / 100;
+
+  const budget = {
+    total_cost: sanitizeBudgetItem({
+      value: total,
+      currency: 'BGN',
+      raw: record['Total Eligible Costs Granted for the Operation'],
+    }),
+    eu_contrib: sanitizeBudgetItem({
+      value: euContrib,
+      currency: 'BGN',
+      raw: euContrib,
+    }),
+    private_fund: sanitizeBudgetItem(),
+    public_fund: sanitizeBudgetItem(),
+    other_contrib: sanitizeBudgetItem(),
+    funding_area: [],
+    mmf_heading: '',
+  };
+
+  return budget;
+};
+
+/**
+ * Preprocess `description` field.
+ *
+ * Input fields taken from the `record` are:
+ * - `Summary of the Operation`
+ *
+ * @memberof BulgariaXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getProjectDescription = record =>
+  record['Summary of the Operation'] || '';
+
+/**
+ * Preprocess `third_parties` field.
+ *
+ * Input fields taken from the `record` are:
+ * - `Beneficiary Name`
+ *
+ * @memberof BulgariaXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array<ThirdParty>}
+ */
+
+const getThirdParties = record => {
+  const beneficieries = [];
+
+  if (record['Beneficiary Name']) {
+    const beneficiary = {
+      address: '',
+      country: '',
+      email: '',
+      name: record['Beneficiary Name'],
+      phone: '',
+      region: '',
+      role: '',
+      type: 'beneficiary',
+      website: '',
+    };
+
+    beneficieries.push(beneficiary);
+  }
+
+  return beneficieries;
+};
+
+/**
+ * Generates `project_id`.
+ *
+ * Input fields taken from the `record` are:
+ * - `Operation Name`
+ *
+ * @memberof BulgariaXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getProjectId = record =>
+  crypto
+    .createHash('md5')
+    .update(record['Operation Name'])
+    .digest('hex');
+
+/**
+ * Preprocess `project_locations`.
+ *
+ * Input fields taken from the `record` are:
+ *
+ * - `Location`
+ *
+ * @memberof BulgariaXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Array<Location>}
+ */
+const getLocations = record => {
+  const locations = [];
+
+  if (record.Location) {
+    const locs = record.Location.split(';').filter(l => l);
```

Comment:
shouldn't we at least push 1 location with `country_code: 'BG'` only in any case? all those project happened in Bulgaria, right? or are there exceptions?

Reply:
Default location added.